src/Microsoft.Azure.SignalR.Common/Endpoints/AccessKeyManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class AccessKeyManager : IAccessKeyManager
+    {
+        private readonly TimerAwaitable _timer = new TimerAwaitable(TimeSpan.Zero, TimeSpan.FromMinutes(5));
+
+        private readonly ConcurrentDictionary<HubServiceEndpoint, object> _endpoints = new ConcurrentDictionary<HubServiceEndpoint, object>();
```

Comment:
you might need a custom IEqualityComparer for the dict here, `ServiceEndpoint` overrides `GetHashCode` on purpose

Reply:
And please add UT to cover it